### ISSUE TITLE
Add part attribute to inner elements

### ISSUE THREAD

**Issue by walidaalhomsi** [add2cal/add-to-calendar-button#545]
Hello,

Thank you for the awesome library,

I think adding `part` attribute would provide us with styling freedom so we can target the inner element with css rules:

```css
add-to-calendar-button::part(atcb-button) {
    /** our rules here **/
}
```

It can be easily done in the library implementation by adding corresponding `part` attribute which should be equilevent to the element `class`.

For example, you can add the following code after this [**line**](https://github.com/add2cal/add-to-calendar-button/blob/5e718b5b3cf8086524a1e65c1c3e4682a4be6490/src/atcb-generate.js#L206):

```javascript
buttonTrigger.setAttribute('part', 'atcb-button')
```

Let me know if I can submit a PR for this ...

**Comment by jekuer** [comment]
Hi @walidaalhomsi ,

Sure. We always love PR! 😃 
We might then extend it a little bit.
We should at least provide `part` attributes for the following elements:

- atcb-button-wrapper
- atcb-button
- atcb-button-icon
- atcb-button-text
- atcb-list-wrapper
- atcb-list
- atcb-list-item
- atcb-list-icon
- atcb-list-text
- atcb-modal-box

As this got lately mentions at #541, are you working together?

Best,
Jens

**Comment by walidaalhomsi** [comment]
Hello @jekuer 

Thank you for your reply, and sorry for my late response,

I will do the PR within a few weeks from now inshallah,

And no I'm not working with the guys from #541  :)

**Comment by jekuer** [comment]
Implemented with v2.6

**Comment by walidaalhomsi** [comment]
Thank you so much @jekuer sorry I didn't have a chance to do a PR ...
